### PR TITLE
Make it "Definitely Typed" and revise docs on Definition Owners

### PR DESCRIPTION
**Pull request by orta**
This is README only changes, to give a better explanation about what a Definition Owner is and how you can be one

### CODE REVIEW

File: README.md
Comment (sandersn) [inline README.md:245]: Is it possible to query github's recent editors to figure this out? But I guess since we now use the codeowners list to decide who can sign off and publish, the recent editors list would be too permissive.

File: README.md
Comment (sandersn) [inline README.md:70]: this feels really petty, but I actually do care about spelling the product "Definitely Typed" and only having the path be "DefinitelyTyped". And there aren't any international trademarks that I'm aware of, unlike JavaScript and MicroSoft.

Can you make this change in a separate PR and we can discuss there?

**Review by sandersn — APPROVED** [review]
Looks good except for DefinitelyTyped being only one word.

File: README.md
Comment (orta) [reply README.md:70]: I can, but FWIW, that's how the spelling has always been in the docs/site for DT: http://definitelytyped.org/guides/pull-request.html

File: README.md
Comment (orta) [reply README.md:245]: Yeah, exactly that was the goal - this requires a bit more friction, but considering that gives kinda-write access it's worth the friction

File: README.md
Comment (sheetalkamat) [reply README.md:70]: I am assuming we dont want this spelling change in this PR.

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

**Comment by orta** [comment]
Alright, all the naming is resolved - for anyone watching along:

> "DefinitelyTyped" is the repo name only, which you basically never need to refer to - "Definitely Typed" is the name of Definitely Typed 🍡 

This is based on some historic notes / emails in https://blog.johnnyreilly.com/2019/10/definitely-typed-movie.html and some form of consensus in the TS team, I just made it consistent everywhere I this repo.

File: README.md
Comment (alloy) [inline README.md:232]: Thanks! Hope this is going to be enough to stop people adding themselves too soon 🤞

Comment (simonbuchan) [reply README.md:232]: Yeah, I'm curious what, if anything, should be done for existing packages. In practice on `aws-lambda` I've mostly just seen that most pinged users never respond, so I'm not that concerned, but maybe they're all just quietly fuming at the spam?

Comment (jagregory) [reply README.md:232]: As one of those `aws-lambda` folks, I'm happy for the spam, though you seem to smash through the requests @simonbuchan before I ever get around to them. 😃

Comment (simonbuchan) [reply README.md:232]: Having an opinion had never been a problem for me! But feel free to toss in a second review, even if it's just a empty accept: I'm a bit concerned about my random quirks being gospel for such a popular package.

Comment (orta) [reply README.md:232]: Yeah, it's not much different from now than before (though maybe because PRs don't need to be delayed by us getting round to it, maybe there could be more) 

Part of our internal chat which trigger this PR covered idle module owners - IMO, I think they're fine to stay as that, but there's probably a space to include docs on how to remove yourself. It's hard though because putting that on a new PR's chat messages feels a bit weird, but maybe it could be in this message  https://github.com/DefinitelyTyped/DefinitelyTyped/pull/44942#issuecomment-631774581 as a disclosure section 

e.g. 

<details>
    <summary>Getting too many notifications? </summary>
    how to remove yourself
</details>

Comment (simonbuchan) [reply README.md:232]: Yeah, that seems reasonable.
The other case the had been a problem I've seen was people opening issues for a package getting daunted by the number of owners and either not mentioning anyone or only picking a few. It would be nice if the bot could handle this, but for now having fewer owners is a decent mitigation. But DT issues are weird in a few ways

Comment (alloy) [reply README.md:232]: @orta If it remains an issue, I think it would be great if the bot could leave a danger-like comment saying “please verify that you intended to sign-up as maintainer and get pinged for each issue/PR”. I did that manually for a while and it seemed to make half the people realize what it meant and saying “oh no thanks”. I stopped doing it, though, too tiresome and I feel bad for letting a PR sit 36 hours because of back and for the across a ~12 hour time zone difference.

**Pull request merged**
